code/ship/shiphit.cpp

```diff
@@ -1783,46 +1783,27 @@ extern int Homing_hits, Homing_misses;
 // Goober5000 - note... hit_pos is in *local* coordinates
 void ship_apply_whack(vec3d *force, vec3d *hit_pos, object *objp)
 {
+	Assertion((objp != nullptr) && (force != nullptr) && (hit_pos != nullptr), "ship_apply_whack invalid argument(s)");
+
 	if (objp == Player_obj) {
 		nprintf(("Sandeep", "Playing stupid joystick effect\n"));
 		vec3d test;
 		vm_vec_unrotate(&test, force, &objp->orient);
 
 		game_whack_apply( -test.xyz.x, -test.xyz.y );
 	}
+	
+	vec3d world_hit_pos;
+	vm_vec_unrotate(&world_hit_pos, hit_pos, &objp->orient);
```

Comment:
Again it looks like this is an incomplete formula.  To go from local to global, you do an unrotate followed by an add.  To go from global to local, you do a subtract followed by a rotate.  These operations must always be paired to obtain the complete transformation.